### PR TITLE
Grille Code Update

### PR DESCRIPTION
**Pull request by uraniummeltdown**
Grille code refactor from tgstation, metal rods can no longer be attack_self()'d for grilles they now use recipes, added broken grille object for mapping, grille code uses deconstruct() and take_damage() instead of healthcheck(), broken grilles when attacked with a metal rod can be rebuilt, if you break grilles instead of using wirecutters they drop 1 rod and become broken, breaking a broken grille drops another rod, also general code cleanup. Replaced all the varedited broken grilles with the new object. Rods max_amount 60->50

:cl:
add: You can now damage grilles by throwing things at them.
add: You can rebuild broken grilles by using a metal rod on them.
tweak: Metal rods maximum amount reduced from 60 to 50.
/:cl:

### CODE REVIEW

**Comment by DesolateG** [comment]
Recipes, meaning you can't click on the metal rod and make a grille?

**Comment by uraniummeltdown** [comment]
Nope, that's because I plan on adding more recipes to rods later (table frames instead of table parts)

File: code/game/objects/items/stacks/rods.dm [hunk 1]
```diff
@@ -11,7 +15,7 @@
 	throw_speed = 3
 	throw_range = 7
 	materials = list(MAT_METAL=1000)
-	max_amount = 60
+	max_amount = 50
```
Comment (Krausus) [inline code/game/objects/items/stacks/rods.dm]: This isn't mentioned in the list of changes. :eyes:

Comment (uraniummeltdown) [inline code/game/objects/items/stacks/rods.dm]: I wanted to standardize it as the ores, glass, plasteel and metal all have max 50. Updated changelog.

**Comment by SamHPurp** [comment]
Seems sensible, but it is now easier for people to break out of cells with the ability to break the grills by chucking stuff at it. Personally I'm for that, but it is a balance thing to consider.

File: code/game/objects/structures/grille.dm [hunk 2]
```diff
@@ -37,47 +41,48 @@
 	//height=42
 	icon='icons/fence-ns.dmi'
 
-/obj/structure/grille/ex_act(severity)
-	qdel(src)
+/obj/structure/grille/ex_act(severity, target)
```
Comment (Fox-McCloud) [inline code/game/objects/structures/grille.dm]: no `target` argument.

**Pull request merged**